### PR TITLE
docs: add missing arguments to StorageOptions documentation

### PR DESCRIPTION
**Pull request by ddelgrosso1**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/nodejs-storage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: src/storage.ts [hunk 1]
```diff
@@ -484,6 +484,10 @@ export class Storage extends Service {
    *     error should be retried and false otherwise.
    * @property {string} [userAgent] The value to be prepended to the User-Agent
    *     header in API requests.
+   * @property {object} [authClient] GoogleAuth client to reuse instead of creating a new one.
+   * @property {number} [timeout] The amount of time in milliseconds to wait for a http request before timing out.
```
Comment (tritone) [inline src/storage.ts:488]: Is this the total timeout (covering all potential retries in a library call) or per-request?

**Comment by shaffeeullah** [comment]
Where are these parameters actually used? I don't see any tests verifying that they work as expected.

File: src/storage.ts (hunk 1 above)
Comment (shaffeeullah) [reply src/storage.ts:488]: No, this is different.  `totalTimeout` is in `retryOptions`.

Comment (ddelgrosso1) [reply src/storage.ts:488]: Per request, I'll add that caveat to the sentence.

**Comment by ddelgrosso1** [comment]
@shaffeeullah the ones I added are passed into common and primarily used at that level.

**Review by shaffeeullah — APPROVED** [review]

**Pull request merged**